data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -186,6 +208,11 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
                     continue;
                 }
 
+                if (!defaultKeysSet.isEmpty() && defaultKeysSet.contains(key)) {
```

Comment:
This can be simplified into a single condition statement
```suggestion
                if (defaultKeysSet.contains(key)) {
```

Reply:
same applies above on `excludeKeysSet`